main.py

```diff
@@ -37,34 +36,86 @@ def getEcommerceWebsiteName():
     )
     return ecommerce_website_provider
 
-def main():
-    print("Running application...")
-    fake = Faker()
+def addProviders(fake):
     # Add the product category provider
-    fake.add_provider(getProductCategoryProvider())
+    fake.add_provider(products.getProductCategoryProvider())
     fake.add_provider(getPaymentProvider())
     fake.add_provider(getFailureProvider())
     
     fake.add_provider(getEcommerceWebsiteName())
 
 
-    product_category = fake.category()
+    fake.add_provider(products.getBeautyProvider())
+    fake.add_provider(products.getBooksProvider())
+    fake.add_provider(products.getElectronicsProvider())
+    fake.add_provider(products.getFineArtsProvider())
+    fake.add_provider(products.getHealthProvider())
+    fake.add_provider(products.getHomeGardenProvider())
+    fake.add_provider(products.getMusicProvider())
+    fake.add_provider(products.getInstrumentsProvider())
+    fake.add_provider(products.getOfficeProvider())
+    fake.add_provider(products.getOutdoorProvider())
+    fake.add_provider(products.getPetProvider())
+    fake.add_provider(products.getSoftwareProvider())
+    fake.add_provider(products.getSportsProvider())
+    fake.add_provider(products.getToyGamesProvider())
+    fake.add_provider(products.getVideoGamesProvider())
+    fake.add_provider(products.getVideoProvider())
+    fake.add_provider(products.getWatchProvider())
+
+def appendGeneratedData(fake, data):
+    category_product_id = products.grab_random_category_product(fake).split(',')
+
+    order_id = generateOrderId()
+    customer_id = fake.unique.random_int(min=111111, max=999999)
+    customer_name = generate_name(fake)
+    product_id = category_product_id[2]
+    product_name = category_product_id[0]
+    product_category = category_product_id[1]
     payment_type = fake.payment_type()
     quantity = fake.pyint(min_value=1, max_value=20)
     price = round(fake.pyfloat(min_value=1.00, max_value=200.00), 2)
-    failure_reason = fake.failure_reason()
-
     date_time = fake.date_time().strftime("%Y-%m-%d %H:%M")
-    ecommerce_website = fake.ecommerce_website()
     country = fake.country()
     city = fake.city()
+    ecommerce_website = fake.ecommerce_website()
     payment_txn_id = fake.random_number(digits=5, fix_len=True)
     payment_txn_success = fake.random_element(elements=['Y', 'N'])
+    failure_reason = fake.failure_reason()
+    data.append({'order_id': order_id, 'customer_id': customer_id, 'customer_name': customer_name, 'product_id': product_id, 
+                    'product_name': product_name, 'product_category':product_category, 'payment_type':payment_type, 'qty':quantity, 'price':price, 
+                    'datetime':date_time, 'country':country, 'city':city, 'ecommerce_website_name':ecommerce_website, 'payment_txn_id':payment_txn_id,
+                    'payment_txn_success': payment_txn_success, 'failure_reason': failure_reason})
```

Comment:
Sanity check: Does this produce the expected outcome? Looking at main.data, it's a list, but then here it's appending a dictionary to that list. If csv module is anticipating a dictionary and just grabs the keys from it, then this comment is good to be ignored, otherwise I'd recommend regenerating the csv file to make sure that the output is what we anticipate